### DISCUSSION TITLE
Refetch on focus interval

### DISCUSSION THREAD

**Discussion by clicktodev** [TanStack/query#2482] [Ideas]
Currently if a request returns an error then it will infinitely refetch on focus if it keeps returning an error.
Is there a way to add an interval so that for example the refetch on focus retries every 5 minutes.
Perhaps only add the interval to failed requests.

**Comment by TkDodo** [comment]
> Currently if a request returns an error then it will infinitely refetch on focus if it keeps returning an error.

why infinitely? It should only fetch for how long you have `retry` specified, which is 3 times per default.

> Is there a way to add an interval so that for example the refetch on focus retries every 5 minutes.

you can use the `refetchInterval` option to do an interval fetching

> Perhaps only add the interval to failed requests.

For that, you'd need to turn the refetchInterval on/off depending on the query state, which you'd need likely need to capture in local state. something like:

```
const [hasError, setHasError] = useState(false)
useQuery(
  key,
  fn,
  {
    onSettled: (data, error) => setHasError(!!error)
    refetchInterval: hasError ? 5 * 60 * 1000 : 0
  }
```
